infoTree/scrapeLeaves.py

```diff
@@ -0,0 +1,175 @@
+from bs4 import BeautifulSoup as BS
+import requests
+from builtins import type
+from bs4.element import NavigableString
+from bs4.element import Tag
+import os
+
+replace = {
+    'a' : '"link"',
+    'h3' : '"heading"',
+    'p' : '"para"',
+    'strong' : '"bold"',
+    'h1': '"title"',
+    'li': '"bullet"',
+    'ul': '"list"',
+    'div': '"CONTAINER"'
+    
+}
+
+class Scrape(object):
+    """docstring for ClassName"""
+    def __init__(self, URL, count, debug):
+        self.indent = 0
+        self.outfile = open('out/url' + str(count) + '.txt', 'w')
+        self.count = count
+        self.URL = URL
+        self.debug = debug
+
+    def Solve(self):
+        print('scraping URL' + str(self.count) + ': ' + self.URL + '\n')
+        scrape = self.URL.strip('\n')
+        
+        page = requests.get(scrape)
+        soup = BS(page.content, "html.parser")    
+            
+        root = soup.find(id = 'region')
+        
+        self.outfile.write('{\n')
+        self.indent = self.indent + 1
+        self.write_Indent('"URL' + str(self.count) + '":' ,None,'','',True)
+
+
+        try :
+            self.write_tree_into_JSON(root, True)
+
+        except Exception as ex:
+            print(ex)
+            file = open('errog.txt', 'a')
+            file.write('error with URL ' + str(self.count) + '\n')
+            file.close()
+        
+        self.outfile.close()
+        return self.count + 1
+        
+    def ParseNavStr(self,elt):
+        toConvert = str(elt.string)
+        return str(toConvert.encode('ascii', 'replace').decode("utf-8"))
```

Comment:
I wonder if I have to configure the writing to output file a certain way... this code may not be useful, but it does prevent the code from crashing as it crashes trying to write the weird chars into the output file. Did your friend have any insight on this?

Reply:
He said Python3 should take care of that kind of stuff for you, but I didn't dig into the error you saw with him. Is it just on file write?

Reply:
codec can't encode character u'\u2013' in position 9: ordinal not in range(128)

This was the type of error i was getting upon file write, and yes it was only on file write

Reply:
I'm sure this issues is solved somewhere on the internet https://stackoverflow.com/questions/9942594/unicodeencodeerror-ascii-codec-cant-encode-character-u-xa0-in-position-20 
seems to indicate that you should swap out `str()` which can't deal with that, with `.encode('utf-8')`. I don't know what you've already tried though

Reply:
I just made the output file encoded in "utf-8" rather than "ascii", which solves the problem. I guess this should have been an obvious fix, but this didn't occur to me until after I looked into it more